crates/client-api/src/routes/database.rs

```diff
@@ -92,23 +92,37 @@ pub async fn call<S: ControlStateDelegate + NodeDelegate>(
             host.spawn_module_host(dbic).await.map_err(log_and_500)?
         }
     };
-    let result = match module.call_reducer(caller_identity, None, &reducer, args).await {
-        Ok(rcr) => rcr,
-        Err(e) => {
-            let status_code = match e {
-                ReducerCallError::Args(_) => {
-                    log::debug!("Attempt to call reducer with invalid arguments");
-                    StatusCode::BAD_REQUEST
-                }
-                ReducerCallError::NoSuchModule(_) => StatusCode::NOT_FOUND,
-                ReducerCallError::NoSuchReducer => {
-                    log::debug!("Attempt to call non-existent reducer {}", reducer);
-                    StatusCode::NOT_FOUND
+
+    let result = if let Err(e) = module.call_identity_connected_disconnected(caller_identity, true).await {
+        return Err((StatusCode::NOT_FOUND, format!("{:#}", anyhow::anyhow!(e))).into());
```

Comment:
I'm not sure `StatusCode::NOT_FOUND` is appropriate here, tho I don't really know what will cause `call_identity_connected_disconnected` to return `Err`.

Reply:
It appears to only return an error on `ModuleNotFound`, I figured this was appropriate 🤷‍♂️